drivers/adc/ad463x/ad463x.c

```diff
@@ -100,6 +100,31 @@ int32_t ad463x_spi_reg_write(struct ad463x_dev *dev,
 	return no_os_spi_write_and_read(dev->spi_desc, buf, 3);
 }
 
+/**
+ * @brief SPI read device register using a mask.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - The mask.
+ * @param data - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad463x_spi_reg_read_masked(struct ad463x_dev *dev,
+				    uint16_t reg_addr,
+				    uint8_t mask,
+				    uint8_t *data)
+{
+	uint8_t reg_data;
+	int32_t ret;
+
+	ret = ad463x_spi_reg_read(dev, reg_addr, &reg_data);
```

Comment:
shouldn't we pass `reg_data` together with the mask to the `data` pointer? or as I mentioned before, we can use only `data` pointer for both reading from the register and applying mask.